export.js

```diff
@@ -0,0 +1,161 @@
+function grabExcelValues(rows, artifact, objTemplate, customFieldRange) {
+    if (rows === null) {
+        getRows(artifact, objTemplate, customFieldRange);
+    }
+    else {
+        return Excel.run(function (context) {
+            let sheet = context.workbook.worksheets.getItem(toSheetName(artifact));
+            let inputRange = columnRanges[artifact] + rows;
+            let customRange = (customFieldRange[0] + 3) + ":" + (customFieldRange[1] + rows);
+            let inputValues = sheet.getRange(inputRange);
+            let customFields = sheet.getRange(customRange);
+            inputValues.load();
+            customFields.load();
+            return context.sync()
+                .then(function () {
+                    customFieldObjArr = customFields.values.map(customFieldObjCreate);
+                    ajaxExport(inputValues.values, artifact, objTemplate, customFieldObjArr);
+                })
+        });
+    }
+}
+
+function getRows(artifact, objTemplate, customFieldRange) {
```

Comment:
it does use getRows to get the rows it needs, it's circular because of the way excel works but I'll look into reworking it. Right now grabExcelValues gets called with no rows, sees that it has no rows, calls getRows, then get rows sends you back into grabExcelValues with the correct rows. Definitely confusing though